sql/size_utils.sql

```diff
@@ -365,6 +365,76 @@ BEGIN
 END
 $BODY$;
 
+CREATE OR REPLACE FUNCTION _timescaledb_internal.data_node_hypertable_approximate_row_count(
+    node_name   NAME,
+    schema_name NAME,
+    table_name  NAME
+)
+RETURNS TABLE (schema_name NAME, table_name NAME, row_estimate BIGINT)
+AS '@MODULE_PATHNAME@', 'ts_dist_remote_hypertable_approximate_row_count' LANGUAGE C VOLATILE STRICT;
+
+CREATE OR REPLACE FUNCTION _timescaledb_internal.hypertable_remote_approximate_row_count(
+    schema_name_in NAME,
+    table_name_in  NAME
+)
+RETURNS TABLE (schema_name NAME, table_name NAME, row_estimate BIGINT)
+LANGUAGE PLPGSQL VOLATILE AS
+$BODY$
+DECLARE
+BEGIN
+    RETURN QUERY
+	SELECT
+		schema_name_in AS schema_name,
+		table_name_in AS table_name,
+		result.row_estimate::bigint AS row_estimate
+	FROM (
+		SELECT
+			sum(entry.row_estimate) AS row_estimate
+		FROM (
+			SELECT
+				s.node_name
+			FROM
+				_timescaledb_catalog.hypertable AS ht,
+				_timescaledb_catalog.hypertable_data_node AS s
+			WHERE
+				ht.schema_name = schema_name_in
+				AND ht.table_name = table_name_in
+				AND s.hypertable_id = ht.id
+		) AS srv
+		LEFT OUTER JOIN LATERAL
+			_timescaledb_internal.data_node_hypertable_approximate_row_count(srv.node_name, schema_name_in, table_name_in) entry ON TRUE
+		GROUP BY entry.schema_name, entry.table_name
+		ORDER BY entry.schema_name, entry.table_name
+	) result;
+END
+$BODY$;
+
+CREATE OR REPLACE FUNCTION _timescaledb_internal.hypertable_local_approximate_row_count(
+    schema_name_in NAME,
+    table_name_in  NAME
+)
+RETURNS TABLE (schema_name NAME, table_name NAME, row_estimate BIGINT)
+LANGUAGE PLPGSQL VOLATILE AS
+$BODY$
+DECLARE
+BEGIN
+	RETURN QUERY
+	SELECT h.schema_name,
+		h.table_name,
+		row_estimate.row_estimate
+	FROM _timescaledb_catalog.hypertable h
+	CROSS JOIN LATERAL (
+		SELECT sum(cl.reltuples)::BIGINT AS row_estimate
```

Comment:
We have to take compressed chunks into account. The row count before compression is in the compression stats tables.
We have to account for duplicate chunks for the distributed case .

Reply:
We haven't done this before, maybe introduce compression support in separate PR in that case? 

Not really sure about the duplicate chunks case, can you specify? I am not sure we support native replication yet

Reply:
actually with brians patch applied this will get correct stats for compressed and uncompressed chunks